feat: Add can-read and can-write helpers

bertdeblock opened the pull request.

Windvis commented on addon/helpers/can-helper-base.js:
```diff
@@ -0,0 +1,45 @@
+import { bind } from '@ember/runloop'
+import { inject as service } from '@ember/service'
+
+export default Helper.extend({
```
This is just a mixin in disguise 😄, but I don't really know a better solution, so meh.

bertdeblock replied:
Yeah, we briefly discussed this some time ago. Okay like this or should I look for a better solution?

Windvis replied:
Nah, it's ok

bertdeblock replied:
Okay!

Windvis commented on addon/helpers/can-helper-base.js:
```diff
@@ -0,0 +1,45 @@
+    this.recompute = bind(this, this.recompute)
+
+    this.recordPermissionsService.on(
+      'model-permissions-changed',
```
I would maybe put these into "enum" constants?

bertdeblock replied:
Done.

Windvis commented:
Build fails on 2.18 due to native getter usage.

bertdeblock commented:
I think everything should be okay now.

- Removed 2.18 support in a separate commit
- Removed the usage of the `Evented` mixin
- Made sure the helpers only recompute when actually needed

Windvis commented on app/helpers/can-read.js:
```diff
@@ -0,0 +1,4 @@
+export {
+  default,
+  canRead
```
Is no longer exported I think

bertdeblock replied:
SHIT!

Windvis replied:
Yea, easy to overlook, I assume I have the same problem in my helper MR :P Silly that ember-cli can't detect this during a build :/

bertdeblock replied:
Removed.

Windvis commented on app/helpers/can-write.js:
```diff
@@ -0,0 +1,4 @@
+export {
+  default,
+  canWrite
```
same

bertdeblock replied:
Removed.
